### PR TITLE
Mod2gems - move some mod/X Modules to gem_mod and make them gem 2

### PR DESCRIPTION
**Pull request by GerryG**
New PR because I broke the old branch

### CODE REVIEW

File: card/gem_mods/ace_editor/card-mod-ace_editor.gemspec [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+vbits = version.split('.').map &:to_i
+vplus = { 0 => 90, 1 => 100 } # can remove and hardcode after 1.0
+vminor = vplus[ vbits[0] ] + vbits[1]
+card_version = [1, vminor, vbits[2]].compact.map(&:to_s).join "."
+# see card.gemspec for explanation of all of this, which has been ham-handedly
+# cut and pasted here.
+
+Gem::Specification.new do |s|
+  s.name = "card-mod-ace_editor"
+  s.version = card_version
```
Comment (ethn) [inline card/gem_mods/ace_editor/card-mod-ace_editor.gemspec:13]: we actually just want "version", not card_version.  card_version is going to be weird until we get to decko 1.0.  Then every gem will just use the same version.

**Review by ethn — CHANGES_REQUESTED** [review]
This PR is never going to pass (or even really run) CI tests until we take care of template/Gemfile. I think I already explained how to do that, but let me know if you need more help.

File: card/gem_mods/ace_editor/card-mod-ace_editor.gemspec (hunk 1 above)
Comment (GerryG) [reply card/gem_mods/ace_editor/card-mod-ace_editor.gemspec:13]: Ok, I'll change that. I wasn't sure how that would work with the other version streams.

**Comment by GerryG** [comment]
I'm not sure I get the Gemfile issue. I have to change the generator so it makes a good Gemfile in decko new. Is that the gist of it. I guess I need to use that directly in my Docker stuff too. That would test it before it gets to ci. I think I'll try that.

File: decko/decko.gemspec [hunk 2]
```diff
@@ -12,7 +12,7 @@ card_version = [1, vminor, vbits[2]].compact.map(&:to_s).join "."
 
 Gem::Specification.new do |s|
   s.name          = "decko"
-  s.version       = version
+  s.version       = card_version
```
Comment (ethn) [inline decko/decko.gemspec:15]: yikes! no not this.

**Comment by GerryG** [comment]
I think I fixed it and the version change.

File: decko/decko.gemspec (hunk 2 above)
Comment (GerryG) [reply decko/decko.gemspec]: fixed all the versions

File: decko/lib/decko/generators/decko/templates/Gemfile [hunk 3]
```diff
@@ -6,8 +6,24 @@ path "<%= @gemfile_gem_path %>" do
...
   gem 'cardname', :require => false
 end
+path "<%= @gemfile_gem_path %>/card/gem_mods" do
+  gem 'card-mod-date'
+  gem 'card-mod-recaptcha'
+  gem 'card-mod-ace_editor'
+  gem 'card-mod-prosemirror_editor'
+  gem 'card-mod-tinymce_editor'
+  gem 'card-mod-edit'
+  #gem 'card-mod-layout'
```
Comment (ethn) [inline decko/lib/decko/generators/decko/templates/Gemfile:16]: I think we can delete (not just comment) these.

Comment (ethn) [reply decko/lib/decko/generators/decko/templates/Gemfile:16]: (mod-layout only)

File: card/mod/Modfile [hunk 4]
```diff
@@ -21,12 +18,15 @@ mod 'media'
...
 mod 'bootstrap'
 mod 'google_analytics'
-mod 'recaptcha'
+gem_mod 'recaptcha'
 gem_mod 'markdown'
+gem_mod 'date'
+gem_mod 'ace_editor'
+gem_mod 'prosemirror_editor'
+gem_mod 'tinymce_editor'
+gem_mod 'edit'
```
Comment (ethn) [inline card/mod/Modfile:31]: Based on what we learned the other day about how gem_mods are loaded, these may not be necessary?

Comment (GerryG) [reply card/mod/Modfile]: Yeah, not at the end anyway. I just moved them back to the original order.

File: .gitmodules [hunk 5]
```diff
@@ -19,11 +19,11 @@
 [submodule "card/mod/machines/vendor/jquery_rails"]
 	path = card/mod/machines/vendor/jquery_rails
 	url = https://github.com/rails/jquery-rails
-[submodule "card/mod/date/vendor/moment"]
-	path = card/mod/date/vendor/moment
+[submodule "card/gem_mods/date/vendor/moment"]
```
Comment (ethn) [inline .gitmodules:22]: realizing we should singularize `gem_mod` to be consistent with `mod`

Comment (ethn) [reply .gitmodules:22]: also, this should be `card-mod-date`, not just `date`, right?

**Comment by GerryG** [comment]
I think I know what is breaking. I'm not sure about what those tests are using for Gemfile, is that to make sure it will run on a previous configuration (Gemfile). It is supposed to work without referencing the card-mod-X gems?

**Pull request merged**